cpp/ql/lib/semmle/code/cpp/ir/implementation/aliased_ssa/internal/SSAConstruction.qll

```diff
@@ -702,6 +908,54 @@ module DefUse {
     hasDefinition(_, defLocation, defBlock, defOffset) and
     result = getPhi(defBlock, defLocation) and
     actualDefLocation = defLocation
+    or
+    exists(
+      Alias::VariableGroup vg, int index, UninitializedGroupInstruction initGroup,
+      Alias::GroupedMemoryLocation gml
+    |
+      // Add 3 to account for the function prologue:
+      // v1(void)    = EnterFunction
+      // m1(unknown) = AliasedDefinition
+      // m2(unknown) = InitializeNonLocal
+      index = 3 + vg.getInitializationOrder() and
```

Comment:
The instruction ordering / indexing code has accumulated a very high level of complexity for what it is, and I feel this is error prone and potentially difficult to maintain.  I think we should discuss ways we can improve this in future, in particular a generic way to append one fixed or variable length block of instructions to another (such as the 'prologue' + variable group initialization + rest of the instructions here).